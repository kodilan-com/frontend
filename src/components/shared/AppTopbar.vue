<script>
export default {
  data() {
    return {
      isVisible: !!(localStorage.getItem('isTopbarClosed')),
    };
  },
  methods: {
    handleClickClose() {
      this.isVisible = true;
      localStorage.setItem('isTopbarClosed', true);
    },
  },
};
</script>

<template>
  <section class="component is-topbar" v-if="!isVisible">
    <div class="container">
      <div class="topbar--cols">
        <a
          href="https://github.com/kodilan-com/"
          target="_blank"
          class="topbar--cols__elm is-icon"
        >
          <i class="fa fa-github" />
        </a>
        <div class="topbar--cols__elm is-content">
          <p>
            Kodilan.com açık kaynaklı bir girişimdir.
            <a href="https://github.com/kodilan-com/" target="_blank">
              Katkıda bulunun!
            </a>
          </p>
        </div>
        <a
          href="#"
          @click.prevent="handleClickClose"
          title="Bir daha gösterme."
          class="topbar--cols__elm is-action"
        >
          <i class="fa fa-close" />
        </a>
      </div>
    </div>
  </section>
</template>

<style lang="scss" scoped>
  $this: 'topbar';

  section.component {
    &.is-#{$this} {
      background: #231F20;
      color: #fff;

      a {
        &:hover {
          color: #fff;
        }
      }

      p {
        margin: 0;
      }

      .#{$this}--cols {
        display: flex;
        align-items: center;

        &__elm {
          padding: 8px;

          &.is-content {
            flex: 1;

            @media (max-width: 767px) {
              text-align: center;
            }
          }
          &.is-icon {
            font-size: 24px;
          }

          &.is-action {
            font-size: 18px;
          }
        }
      }
    }
  }
</style>
