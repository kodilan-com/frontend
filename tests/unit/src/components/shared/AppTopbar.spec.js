import { shallowMount } from '@vue/test-utils';
import AppTopbar from '@/components/shared/AppTopbar';

describe('AppTopbar.vue', () => {
  let wrapper;


  beforeEach(() => {
    wrapper = shallowMount(AppTopbar);
  });

  afterEach(() => {
    wrapper.destroy();

    jest.resetAllMocks();
  });

  it('should match with snapshot, if not clicked the "Bir daha gösterme." button before', () => {
    localStorage.getItem.mockReturnValue(undefined);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should match with snapshot, if clicked the "Bir daha gösterme." button before', () => {
    localStorage.getItem.mockReturnValue(true);

    wrapper = shallowMount(AppTopbar);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not show the topbar section, if clicked the "Bir daha gösterme." button', () => {
    localStorage.getItem.mockReturnValue(undefined);

    wrapper.find('a[title="Bir daha gösterme."]').trigger('click');

    expect(wrapper.find('.is-topbar').exists()).toBe(false);
    expect(localStorage.setItem).toHaveBeenCalledWith('isTopbarClosed', true);
  });
});
