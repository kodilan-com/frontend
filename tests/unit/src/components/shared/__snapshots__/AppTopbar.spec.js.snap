// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTopbar.vue should match with snapshot, if clicked the "Bir daha gösterme." button before 1`] = `<!---->`;

exports[`AppTopbar.vue should match with snapshot, if not clicked the "Bir daha gösterme." button before 1`] = `
<section
  class="component is-topbar"
>
  <div
    class="container"
  >
    <div
      class="topbar--cols"
    >
      <a
        class="topbar--cols__elm is-icon"
        href="https://github.com/kodilan-com/frontend"
        target="_blank"
      >
        <i
          class="fa fa-github"
        />
      </a>
       
      <div
        class="topbar--cols__elm is-content"
      >
        <p>
          
          Kodilan.com açık kaynaklı bir girişimdir.
          
          <a
            href="https://github.com/kodilan-com/frontend"
            target="_blank"
          >
            
            Katkıda bulunun!
          
          </a>
        </p>
      </div>
       
      <a
        class="topbar--cols__elm is-action"
        href="#"
        title="Bir daha gösterme."
      >
        <i
          class="fa fa-close"
        />
      </a>
    </div>
  </div>
</section>
`;
